types/mapbox__point-geometry/index.d.ts

```diff
@@ -0,0 +1,34 @@
+// Type definitions for @mapbox/point-geometry 0.1
+// Project: https://github.com/mapbox/point-geometry
+// Definitions by: Mathieu Maes <https://github.com/webberig>
+//                 Harel Mazor <https://github.com/HarelM>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare class Point {
+    constructor(x: number, y: number);
+    clone: () => Point;
+    add: (p: Point) => Point;
+    sub: (p: Point) => Point;
+    multiByPoint: (p: Point) => Point;
+    divByPoint: (p: Point) => Point;
+    mult: (k: number) => Point;
+    div: (k: number) => Point;
+    rotate: (k: number) => Point;
+    rotateAround: (k: number, p: Point) => Point;
+    matMult: (m: number[]) => Point;
+    unit: () => Point;
+    perp: () => Point;
+    round: () => Point;
+    mag: () => number;
+    equals: (other: Point) => boolean;
+    dist: (p: Point) => number;
+    distSqr: (p: Point) => number;
+    angle: () => number;
+    angleTo: (b: Point) => number;
+    angleWith: (b: Point) => number;
+    angleWithSep: (x: number, y: number) => number;
+
+    static convert: (a: number[] | Point) => Point;
```

Comment:
what about this line:
```suggestion
    static convert<T extends unknown>(a: T): T extends number[] ? Point : T extends Point ? Point : T;
```
`a` is required, but depending on input, output is different:
https://github.com/mapbox/point-geometry/blob/master/index.js#L296. When T is a point, T could be returned to outline identity.

Reply:
That's probably more accurate... Thanks!